CMakeLists.txt

```diff
@@ -559,6 +560,13 @@ if(BUILD_VINEYARD_SERVER)
                                            nlohmann_json::nlohmann_json
                                            TBB::tbb
     )
+    if(ARROW_SHARED_LIB)
+        target_link_libraries(vineyardd PUBLIC ${ARROW_SHARED_LIB})
+    else()
+        target_link_libraries(vineyardd PUBLIC ${ARROW_STATIC_LIB})
+    endif()
+
+    target_include_directories(vineyardd PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/ctti/include>)
```

Comment:
Why requires ctti in the vineyardd?

Reply:
```
[ 65%] Building CXX object CMakeFiles/vineyardd.dir/src/server/async/ipc_server.cc.o
In file included from /root/Desktop/v6d/v6d/src/client/ds/object_factory.h:27,
                 from /root/Desktop/v6d/v6d/src/client/ds/i_object.h:23,
                 from /root/Desktop/v6d/v6d/src/client/client.h:32,
                 from /root/Desktop/v6d/v6d/modules/io/io/io_factory.h:25,
                 from /root/Desktop/v6d/v6d/src/server/util/spill_file.h:27,
                 from /root/Desktop/v6d/v6d/src/server/memory/usage.h:32,
                 from /root/Desktop/v6d/v6d/src/server/memory/memory.h:46,
                 from /root/Desktop/v6d/v6d/src/server/async/ipc_server.h:29,
                 from /root/Desktop/v6d/v6d/src/server/async/ipc_server.cc:16:
/root/Desktop/v6d/v6d/src/common/util/typename.h:33:10: fatal error: ctti/detail/name_filters.hpp: No such file or directory
   33 | #include "ctti/detail/name_filters.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Without ctti, the compile this complain about this

Reply:
You don't need to include the following headers in your `spill_file.h`,

```
#include "io/io/i_io_adaptor.h"
#include "io/io/io_factory.h"
#include "io/io/local_io_adaptor.h"
```

As you already have a file io adaptor in your server and you don't use anything from the client side.